frontend/javascripts/oxalis/model/sagas/mapping_saga.ts

```diff
@@ -319,14 +314,8 @@ function* fetchMappings(
   }
 }
 
-function buildMappingObject(
-  mappingName: string,
-  fetchedMappings: APIMappings,
-): [Mapping, Array<number>] {
-  const mappingObject: Mapping = {};
-  // Performance optimization: Object.keys(...) is slow for large objects
-  // keeping track of the keys in a separate array is ~5x faster
-  const mappingKeys = [];
```

Comment:
Did you try to preallocate the array with the correct size here? I did a small microbenchmark and it seems to be about 5x faster to fill the array then:

```js
n = 1500000
console.time("A")
let arr = [];
for (let i = 0; i < n; i++) {
	arr.push(i)
}
console.timeEnd("A")

console.time("B")
arr = new Array(n);
for (let i = 0; i < n; i++) {
	arr[i] = i;
}
console.timeEnd("B")

output:
A: 25.44775390625 ms
B: 5.261962890625 ms
```

not sure whether this would make it worth again to keep the keys as an extra array around?

Reply:
oh, I'm not sure whether the total amount is known apriori, at all? :see_no_evil:

Reply:
Thanks for testing! It would be possible to find out the number of keys beforehand (with an extra iteration), but I don't think the time spent building the keys array is substantial enough to warrant the extra iteration and keeping the extra array around

Reply:
ok, makes sense!